### ISSUE TITLE
Ways to avoid multiple detection's for same object 

### ISSUE THREAD

**Issue by torres07** [AlexeyAB/darknet#6856]
Hello. I'm currently working a task that classes are disjoint and one object (in my case a person) can only be assigned to one class. The following figure i will try to illustrate this:

![det](https://user-images.githubusercontent.com/18540959/96471395-ae19e080-1205-11eb-8615-429869dbc82b.png)

I thought about making some changes,

1. use softmax activation in last output layers, how can i do this? i have looked to .cfg and in wiki configuration and the function is not available. In think this can help because softmax assumes that all classes are disjoint. 

> activation=leaky - activation function after convolution: logistic (by default), loggy, relu, elu, selu, relie, plse, hardtan, lhtan, linear, ramp, leaky, tanh, stair, relu6, swish, mish

but i see this: 

> [softmax] - SoftMax CE (cross entropy) layer - Categorical cross-entropy for multi-class classification

so, i can just append [softmax] after [convolutional] layer?

2. in terms of use of NMS, who can i do thought all classes? i think in this way i could remove bbox duplication's

**Comment by stephanecharette** [comment]
Wouldn't it be easier to have "glasses" as a class by itself, and then you'd have "person" in addition to "glasses"?

**Comment by torres07** [comment]
> Wouldn't it be easier to have "glasses" as a class by itself, and then you'd have "person" in addition to "glasses"?

I've already implemented this approach, now im trying the one who i explained. There are already some works that take this approach and it works well.

**Comment by JoeCool90** [comment]
Try:

1) Chuck more training data at the problem
2) In the iou/do_nms_sort_v2 whatever (sorry, its been a while since I looked at the source code) bits, detect that you have overlap of the two classes and reject the one with the lowest score
3) As @stephanecharette said, train glasses and then doing as point (2) but classifying the people overlapping or not the glasses

Hope that makes sense

**Comment by torres07** [comment]
Thanks @JoeCool90, i will try to modify implementation of nms to exclude overlaps between different classes.

@AlexeyAB can you guide me and appoint where i can do this modification?

in box.c have these functions:

do_nms
do_nms_sort
do_nms_sort_v2
do_nms_obj

what i should modify?

**Comment by JoeCool90** [comment]
Been a little while since I looked.  I think there's multiple functions in multiple places (different options, and perhaps some old code?) so you have to kind of manual trace through from the main function and work out what would be called.

Or run it through a debugger and put some breakpoints to see which function is hit.
